### ISSUE TITLE
Secrets with same name but for different namespaces in the secretGenerator throws exists error

### ISSUE THREAD

**Issue by jerrinss5** [kubernetes-sigs/kustomize#1142]
To reproduce this:

- Create a kustomization.yaml with the following contents:
```
---
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

secretGenerator:
- literals:
  - a=b
  - c=d
  name: foobar
  namespace: alice
  type: Opaque
- literals:
  - a=b
  - c=d
  name: foobar
  namespace: bob
  type: Opaque
```
Then run:
 `kustomize build .`

It throws the following error: 
`Error: id resid.ResId{gvKind:gvk.Gvk{Group:"", Version:"v1", Kind:"Secret"}, name:"foobar", prefix:"", suffix:"", namespace:"alice"} exists; must merge or replace`

Is this the intended behavior?
I saw that there is also a similar issue https://github.com/kubernetes-sigs/kustomize/issues/141

**Comment by tkellen** [comment]
/ref https://github.com/kubernetes-sigs/kustomize/issues/1155

**Comment by Liujingfang1** [comment]
This has been fixed by #1221
